datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -121,58 +291,220 @@ where
     }
 }
 
+// given an function that maps a `&str`, `&str` to an arrow native type,
```

Comment:
This is a very cool function. Maybe we can pull it into a shared function/utility at some point